### PR TITLE
As apache2 doesn't hand over the proxy env settings

### PR DESCRIPTION
**Pull request by worldcitizen**
As apache2 doesn't hand over the proxy env settings
These have been "optionally" been added to the obs.conf vhost configuration.
An extra debug line has been added to remote_url.rb

Improvement.
Both BSConfig.pm.template and remote_url.rb now use the settings as entered in /etc/sysconfig/proxy

### CODE REVIEW

File: src/backend/BSConfig.pm.template [hunk 1]
```diff
@@ -358,4 +358,19 @@ our $local_registry_signatures_extension = 1;
 # network name required for services using a local proxy/cache like gemianbox
 # our $obs_service_network = "obs_services";
 
+if (-e '/etc/sysconfig/proxy') {
+  open my $proxysettings, '<', '/etc/sysconfig/proxy' or die "Unable to open file:$!\n";
+  my %hash = map { split /=|\s+/; } <$proxysettings>;
+  close $proxysettings;
+  our $proxy = $proxysettings{HTTP_PROXY};
```
Comment (darix) [inline src/backend/BSConfig.pm.template:365]: shouldnt this be '$hash{'HTTP_PROXY'}`?

**Review by darix — CHANGES_REQUESTED** [review]

File: src/api/app/models/project/remote_url.rb [hunk 2]
```diff
@@ -2,11 +2,21 @@ class Project::RemoteURL
   require 'open-uri'
 
   def self.load(remote_project, path)
+    if File.file?('/etc/sysconfig/proxy')
+      proxysettings = Hash[File.read('/etc/sysconfig/proxy').scan(/(\S+)\s*=\s*"([^"]+)/)]
+      Rails.logger.info "var HTTP_PROXY: #{proxysettings.fetch("HTTP_PROXY")} var HTTPS_PROXY: #{proxysettings.fetch("HTTPS_PROXY")} var NO_PROXY: #{proxysettings.fetch("NO_PROXY")}"
```
Comment (dmarcoux) [inline src/api/app/models/project/remote_url.rb:7]: Using `fetch` will raise a `KeyError` exception if a key isn't present. This should be changed:

```suggestion
      Rails.logger.info "var HTTP_PROXY: #{proxysettings['HTTP_PROXY']} var HTTPS_PROXY: #{proxysettings['HTTPS_PROXY']} var NO_PROXY: #{proxysettings['NO_PROXY']}"
```

I don't know if this is intended, but `FTP_PROXY` is missing in that line.

File: src/api/app/models/project/remote_url.rb [hunk 3]
```diff
@@ -2,11 +2,21 @@ class Project::RemoteURL
...
+      ENV['http_proxy'] = proxysettings.fetch("HTTPS_PROXY") if proxysettings.has_key?("HTTPS_PROXY")
```
Comment (dmarcoux) [inline src/api/app/models/project/remote_url.rb:8]:
```suggestion
      ENV['http_proxy'] = proxysettings['HTTPS_PROXY'] if proxysettings.has_key?("HTTPS_PROXY")
```

File: src/backend/BSConfig.pm.template (hunk 1 above)
Comment (worldcitizen) [reply src/backend/BSConfig.pm.template]: It have improved the code. Should now be OK.

File: src/api/app/models/project/remote_url.rb (hunk 2 above)
Comment (worldcitizen) [reply src/api/app/models/project/remote_url.rb:7]: The code has been fixed and tested.

File: src/api/app/models/project/remote_url.rb (hunk 3 above)
Comment (worldcitizen) [reply src/api/app/models/project/remote_url.rb]: Fixed as request in stead of fetch(), [] used.

**Comment by dmarcoux** [comment]
Be sure to address the issues found by linters. Here's how to do this from the developer wiki: https://github.com/openSUSE/open-build-service/wiki/Development-Environment-Tips-&-Tricks#run-linters

**Comment by irides** [comment]
Hi @worldcitizen, Thanks for your PR, it really helps me.

I'd like to help yout to push this ASAP. I think it's better to use `git rebase` rather than `git merge` in your branch.  Because merge upstream code into your branch seems not a good idea.
In addition, some of the commits are trying to fix your previous commit, which makes it difficult for maintainers to review. So squashing them into one or two commits could be a better choice. This could be done by `git rebase` too.

**Comment by worldcitizen** [comment]
Hi,

How can I rerun this pipeline?

https://app.circleci.com/pipelines/github/openSUSE/open-build-service/5542/workflows/17072528-33b8-40b6-a52c-fdbb53d5621b/jobs/76238

**Comment by irides** [comment]
> Hi, How can I rerun this pipeline?

No need any more.  You can close this PR now because I saw you have created a new one.  That one looks better, except one thing: need a commit title, which is a brief summary of what you do in this commit. Then below the commit title, you can write your message as it was.

**Comment by worldcitizen** [comment]
Created a new pull request.
https://github.com/openSUSE/open-build-service/pull/12084

**Pull request closed without merging**